src/shared/Atlassian.Bitbucket/BitbucketHostProvider.cs

```diff
@@ -183,6 +184,19 @@ public Task StoreCredentialAsync(InputArguments input)
             _context.CredentialStore.AddOrUpdate(credentialKey, credential);
             _context.Trace.WriteLine("Credential was successfully stored.");
 
+            Uri targetUri = GetTargetUri(input);
+            if(IsBitbucketServer(targetUri))
+            {
+                // BBS doesn't usually include the username in the urls which means they aren't included in the GET call, 
+                // which means if we store only with the username the credentials are never found again ...
+                // This does have the potential to overwrite itself for different BbS accounts, 
+                // but typically BbS doesn't encourage multiple user accounts
+                string bbsCredentialKey = GetBbSCredentialKey(input);
```

Comment:
I think we should have the full `BitbucketServer` rather than `BbS`, even if it's a ~mouth~ screenful.. 😄 

```suggestion
                string bbsCredentialKey = GetBitbuckerServerCredentialKey(input);
```

Reply:
It might also be worth a unit test or two on the behaviour or storing two credentials in the case of BB server?